decision_service/dsclientlib/ds_driver_impl.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "logger/ds_logger.h"
+
+namespace decision_service
+{
+  class ranking_response;
+  class api_status;
+
+  class driver_impl {
+  public:
+    using error_fn = void(*)( const api_status&, void* user_context );
+
+  private:
+    utility::config_collection _configuration;
+    error_callback_fn _error_cb;
+    logger _logger;
+
+  public:
+    
+    int ranking_request(const char* uuid, const char* context, ranking_response& response, api_status* status);
```

Comment:
Should we have =nullptr here?

Reply:
This is an internal class that always gets called with an argument (by driver).  So not as much value in having a default value since its not meant to be used directly but can add it here. 

---
In reply to: [187098631](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187098631) [](ancestors = 187098631)

Reply:
Cool.